kernel/thread.c

```diff
@@ -436,32 +435,106 @@ static inline size_t adjust_stack_size(size_t stack_size)
 	const size_t fuzz = random_val % CONFIG_STACK_POINTER_RANDOM;
 
 	if (unlikely(fuzz * 2 > stack_size)) {
-		return stack_size;
+		return 0;
 	}
 
-	return stack_size - fuzz;
+	return fuzz;
 }
 #if defined(CONFIG_STACK_GROWS_UP)
 	/* This is so rare not bothering for now */
 #error "Stack pointer randomization not implemented for upward growing stacks"
 #endif /* CONFIG_STACK_GROWS_UP */
```

Comment:
Should STACK_GROWS_UP be deprecated? I don;t believe the other cache pointer calculations work in that case.

Reply:
at the moment, lots of things (independent of this PR) probably don't work with upward growing stacks, as this is never a configuration that has been tested.

I would not object to completely removing it, but I think we should do that in another PR. As it stands, we'll get a build error if enabled.